core/modules/flowgen.cc

```diff
@@ -327,16 +342,48 @@ bess::Packet *FlowGen::FillPacket(struct flow *f) {
   if (f->packets_left <= 1)
     tcp_flags |= 0x01; /* FIN */
 
-  *(reinterpret_cast<uint32_t *>(p + 14 + /* IP dst */ 16)) = f->flow_id;
-  *(reinterpret_cast<uint8_t *>(p + 14 + /* IP */ 20 + /* TCP flags */ 13)) =
-      tcp_flags;
-  *(reinterpret_cast<uint32_t *>(p + 14 + /* IP */ 20  + /* Seq No*/ 4)) = htonl(f->next_seq_no);
+  // bess::utils::EthHeader* ethheader =
+  // reinterpret_cast<bess::utils::EthHeader*>(p);
+  bess::utils::Ipv4Header *ipheader =
+      reinterpret_cast<bess::utils::Ipv4Header *>(
+          p + sizeof(bess::utils::EthHeader));
+  bess::utils::TcpHeader *tcpheader =
+      reinterpret_cast<bess::utils::TcpHeader *>(
+          p + sizeof(bess::utils::EthHeader) + sizeof(bess::utils::Ipv4Header));
+
+  if (ip_src_range_ != 0) {
+    uint32_t *ip_src = &(ipheader->src);
+    *ip_src = htonl(ntohl(*ip_src) + f->flow_id % ip_src_range_);
+  }
+
+  if (ip_dst_range_ != 0) {
+    uint32_t *ip_dst = &(ipheader->dst);
+    *ip_dst = htonl(ntohl(*ip_dst) + (f->flow_id % ip_dst_range_));
+  }
+
+  if (port_src_range_ != 0) {
+    uint16_t *tcp_src = &(tcpheader->src_port);
+    *tcp_src = htons(ntohs(*tcp_src) + (f->flow_id % port_src_range_));
+  }
+
+  if (port_dst_range_ != 0) {
+    uint16_t *tcp_dst = &(tcpheader->dst_port);
+    *tcp_dst = htons(ntohs(*tcp_dst) + (f->flow_id % port_dst_range_));
```

Comment:
I had considered this but I decided it wasn't too confusing -- but I guess maybe it is counterintuitive and that's bad. Do you know of a fast/easy way to make it randomize the space a bit better? Hash the counter and then pull the offset values from different indices in the hash output?